picard/script.py

```diff
@@ -70,13 +77,32 @@ class ScriptSyntaxError(ScriptParseError):
     pass
 
 
-class ScriptUnknownFunction(ScriptError):
+class ScriptUnknownFunction(ScriptParseError):
     pass
 
 
 class ScriptRuntimeError(ScriptError):
-    def __init__(self, message, name, line, position):
-        super().__init__("${1:s}:{2:d}:{3:d}: {0:s}".format(message, name, line, position))
+    def __init__(self, message, functionstackitem):
+        super().__init__(
+            "{0}:{1}:${2}: {3}".format(
```

Comment:
Please use named arguments:

```python
"{line}:{column}:${name}: {message}".format(
    line=functionstackitem.line,
    column=functionstackitem.column,
    name=functionstackitem.name,
    message=message
)
```

Also is there a special reason to not have `$` in the name of the script function? Documentation about scripting says:
`Functions start with a dollar sign and end with an argument list enclosed in parentheses (e.g. $lower(...)).`
So `$` is definitively a part of the function name.

So we could perhaps do:

```python
class FunctionStackItem(StackItem):
    def __init__(self, line, column, name):
        super().__init__(line, column)
        self.name = '$' + name
```

and get rid of `$` in related format strings. What do you think?

Reply:
I couldn't think of any reason to not have '$'  as part of the function name in the class, so I've added it as suggested.